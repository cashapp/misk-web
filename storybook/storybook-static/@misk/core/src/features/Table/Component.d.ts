export interface ITableProps {
    data: any[];
    range?: number[];
}
export declare const Table: (props: ITableProps) => JSX.Element;
