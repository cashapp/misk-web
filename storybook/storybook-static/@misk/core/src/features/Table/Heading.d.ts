export declare const Heading: (props: {
    data: any;
}) => JSX.Element;
