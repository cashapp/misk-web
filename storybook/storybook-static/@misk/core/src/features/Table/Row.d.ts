export declare const Row: (props: {
    data: any;
    index?: number;
}) => JSX.Element;
