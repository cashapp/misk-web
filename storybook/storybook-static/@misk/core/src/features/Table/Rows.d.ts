export declare const Rows: (props: {
    data: any;
    range: number[];
}) => JSX.Element;
