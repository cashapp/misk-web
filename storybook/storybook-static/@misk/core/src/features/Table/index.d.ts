export * from "./Component";
export * from "./Heading";
export * from "./Row";
export * from "./Rows";
