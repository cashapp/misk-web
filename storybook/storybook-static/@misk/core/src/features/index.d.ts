export * from "./Navbar";
export * from "./Table";
