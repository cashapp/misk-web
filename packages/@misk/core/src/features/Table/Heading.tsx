import * as React from "react"

export const Heading = (props: { data: any }) => {
  const { data } = props
  return (
    <thead>
      <tr>
        <th key={"index"}>{""}</th>
        {Object.entries(data).map(([k]) => (
          <th key={k}>{k}</th>
        ))}
      </tr>
    </thead>
  )
}
