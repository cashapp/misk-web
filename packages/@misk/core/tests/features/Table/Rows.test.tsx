import React from "react"
import { cleanup, render } from "@testing-library/react"
import { Rows } from "src/features/Table"
import { cars } from "./testFixtures"

describe("Table Rows", () => {
  afterEach(cleanup)
  it("Rows can render", () => {
    const { asFragment } = render(<Rows data={cars} range={[0, cars.length]} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("Rows can render with maxRows = 2", () => {
    const { asFragment } = render(<Rows data={cars} range={[0, 2]} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
