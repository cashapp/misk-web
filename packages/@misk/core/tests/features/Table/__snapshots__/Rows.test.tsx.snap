// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Rows Rows can render 1`] = `
<DocumentFragment>
  <tbody>
    <tr>
      <td>
        26f7ab1f-4060-497c-9a7e-a772c2bc1e50
      </td>
      <td>
        Buick
      </td>
      <td>
        LaCrosse
      </td>
      <td>
        2011
      </td>
      <td>
        41
      </td>
      <td>
        €21530,26
      </td>
      <td>
        Integrated even-keeled array
      </td>
      <td>
        41.3006835
      </td>
      <td>
        -8.4787949
      </td>
      <td>
        4/17/2018
      </td>
      <td>
        3:19:26.000
      </td>
      <td>
        6:44:28.000
      </td>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        2545bae7-dcf7-4082-ad9c-9e68a68d579a
      </td>
      <td>
        Oldsmobile
      </td>
      <td>
        Silhouette
      </td>
      <td>
        2003
      </td>
      <td>
        18
      </td>
      <td>
        $25266.44
      </td>
      <td>
        Horizontal exuding framework
      </td>
      <td>
        46.1854516
      </td>
      <td>
        30.3414908
      </td>
      <td>
        3/21/2018
      </td>
      <td>
        15:11:57.000
      </td>
      <td>
        13:07:56.000
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        2f0909dd-1298-4469-ac86-e96269009ee8
      </td>
      <td>
        Mercedes-Benz
      </td>
      <td>
        Sprinter 3500
      </td>
      <td>
        2011
      </td>
      <td>
        88
      </td>
      <td>
        ¥36824.84
      </td>
      <td>
        Ameliorated human-resource neural-net
      </td>
      <td>
        14.5975238
      </td>
      <td>
        -91.7270911
      </td>
      <td>
        4/15/2018
      </td>
      <td>
        9:55:18.000
      </td>
      <td>
        2:45:01.000
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        e76594d6-e59c-4cec-ae98-560a41b7066e
      </td>
      <td>
        Ford
      </td>
      <td>
        Mustang
      </td>
      <td>
        2007
      </td>
      <td>
        65
      </td>
      <td>
        $24100.94
      </td>
      <td>
        Seamless needs-based policy
      </td>
      <td>
        31.129098
      </td>
      <td>
        120.839842
      </td>
      <td>
        4/4/2018
      </td>
      <td>
        5:22:24.000
      </td>
      <td>
        14:04:49.000
      </td>
    </tr>
    <tr>
      <td>
        4
      </td>
      <td>
        a4145f72-a953-4c1d-8cdd-1b4dd9fc7e1a
      </td>
      <td>
        Lexus
      </td>
      <td>
        GX
      </td>
      <td>
        2008
      </td>
      <td>
        73
      </td>
      <td>
        $8585.63
      </td>
      <td>
        User-centric 24/7 attitude
      </td>
      <td>
        -20.5360441
      </td>
      <td>
        29.2814687
      </td>
      <td>
        11/11/2018
      </td>
      <td>
        4:55:37.000
      </td>
      <td>
        5:22:04.000
      </td>
    </tr>
    <tr>
      <td>
        5
      </td>
      <td>
        90688155-b258-408b-b5af-d1631101a119
      </td>
      <td>
        Audi
      </td>
      <td>
        RS 4
      </td>
      <td>
        2008
      </td>
      <td>
        3
      </td>
      <td>
        $19244.95
      </td>
      <td>
        Open-source methodical knowledge user
      </td>
      <td>
        -8.6852273
      </td>
      <td>
        120.409835
      </td>
      <td>
        5/29/2018
      </td>
      <td>
        11:54:06.000
      </td>
      <td>
        20:53:58.000
      </td>
    </tr>
  </tbody>
  0
</DocumentFragment>
`;

exports[`Table Rows Rows can render with maxRows = 2 1`] = `
<DocumentFragment>
  <tbody>
    <tr>
      <td>
        26f7ab1f-4060-497c-9a7e-a772c2bc1e50
      </td>
      <td>
        Buick
      </td>
      <td>
        LaCrosse
      </td>
      <td>
        2011
      </td>
      <td>
        41
      </td>
      <td>
        €21530,26
      </td>
      <td>
        Integrated even-keeled array
      </td>
      <td>
        41.3006835
      </td>
      <td>
        -8.4787949
      </td>
      <td>
        4/17/2018
      </td>
      <td>
        3:19:26.000
      </td>
      <td>
        6:44:28.000
      </td>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        2545bae7-dcf7-4082-ad9c-9e68a68d579a
      </td>
      <td>
        Oldsmobile
      </td>
      <td>
        Silhouette
      </td>
      <td>
        2003
      </td>
      <td>
        18
      </td>
      <td>
        $25266.44
      </td>
      <td>
        Horizontal exuding framework
      </td>
      <td>
        46.1854516
      </td>
      <td>
        30.3414908
      </td>
      <td>
        3/21/2018
      </td>
      <td>
        15:11:57.000
      </td>
      <td>
        13:07:56.000
      </td>
    </tr>
  </tbody>
  0
</DocumentFragment>
`;
